### ISSUE TITLE
iframe integration

### ISSUE THREAD

**Issue by jy2008** [RocketChat/Rocket.Chat#9532]
I am looking to integrate RC with my own website.  I saw this iframe integration option. I noticed that in browser console if i enter `Meteor.loginWithPassword('username-or-email', 'your-password');`, it will log the user in. But how can i write it into my main website's code so whenever user login the main site, it will log in automatically to the RC as well ( i embedded RC into my site). 

I also looked https://github.com/RocketChat/Rocket.Chat/issues/2001, it says this:
```
We create a meteor method that we called when the application first loads (
/login ) that would call the following functions in the
client/routes/router.coffee

FlowRouter.route '/login',
name: 'login'
action: ->
readToken = (name) ->
// YOUR CODE HERE
Meteor.call('dojoAuth', readToken(), (err, x) ->
localStorage.setItem('Meteor.userId', x.userId )
localStorage.setItem('Meteor.loginToken', x.token.token )
localStorage.setItem('Meteor.loginTokenExpires', new Date(x.token.when).
toString() )
return FlowRouter.go 'home'
)
On the server side the MeteorMethod would determine if the user should be
authenticated. This could be as easy as setting a localStorage token that
you pass from your main app to the meteor app. This token then gets passed
to meteor and you can check if the token is valid and send the user session
back to the meteor font end.

Meteor.methods({
dojoAuth: function (cookie) {
// GET YOUR USER HERE

Accounts._insertHashedLoginToken(user._id, hashStampedToken);
future.return({
token: stampedLoginToken,
userId: user._id
});
}).run();
```
But i am very confused, where am i supposed to put this code in? my main site JS portion? but this is coffee script.

Can someone give me some detail or even step by step instruction in terms of how to integrate iframe with RC to auto log in the parent site? 

Many thanks in advance!!!

**Comment by hagaymyr** [comment]
@jy2008 Did you manage to accomplish this eventually?

I'm struggling with this as well and can't find a clear solution/approach for this, not here nor in the forum/StackOverflow...

**Comment by kostiag** [comment]
Does anyone have a solution for this? I am trying to figure out what my options are and RC is one of the possibilities (other is mattermost) seamless integration/authentication is a must.

**Comment by hagaymyr** [comment]
I used the iFrame integration mechanism to hack this.

- I created a shallow web-service that acts as a so-called 3rd party authenticator.
- I navigate to the RC by using "http://<ip>:<port>/home?username=johny&password=blabla".

Flow:
1. Upon the RC iFrame API POST call (that queries the so-called 3rd party authenticator) - return an Unauthorized response.
2. Upon the RC iFrame URL GET call (that displays the 3rd party's login form) - return a simple web page with a simple login form that grabs the Username/Password from the URL query-params and automatically sends the form.
3. Upon the RC iFrame URL POST call -
3.a. Using the RC REST API, performs a login call with the credentials and gets the auth-token.
3.b Returns a web page with a JS script that performs the Meteor.loginWithToken(auth-token) client-side call.

This is really an ugly hack, but me being a novice web-developer, this is what i came up with.
We have no security concerns whatsoever, so this hack works for us.
